pandas/core/algorithms.py

```diff
@@ -447,6 +447,8 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> np.ndarray:
     else:
         values = extract_array(values, extract_numpy=True)
 
+    if type(comps).__name__ == "IntegerArray":
```

Comment:
Will this work if you handle it in the extension array `elif` instead of separately up here?
https://github.com/pandas-dev/pandas/blob/3aa84479e54612eb58dd12d2d1f02674acc321c4/pandas/core/algorithms.py#L467-L471

Reply:
so the issue here is that we actually need to implement .isin on the EAs (numeric dtypes); we *already* do this on the internal EAs (datetime, interval etc). 

cc @jbrockmendel

Reply:
> Will this work if you handle it in the extension array `elif` instead of separately up here?
> https://github.com/pandas-dev/pandas/blob/3aa84479e54612eb58dd12d2d1f02674acc321c4/pandas/core/algorithms.py#L467-L471

Hi Andrew,
Are you suggesting something like this:
```Python
 elif is_extension_array_dtype(comps.dtype) or is_extension_array_dtype( 
     values.dtype 
 ): 
     if type(comps).__name__ == "IntegerArray":
         comps = comps._data
     return isin(np.asarray(comps), np.asarray(values)) 
```
I've made a test the performance is good.

Reply:
yeah that was what i meant

Does anything break, and do you get an `isin` speedup,  if you relax the `IntegerArray` check and use `comps._data` for other EAs (such as `FloatingArray`)?

Reply:
Yes, the performance increases from 20ms to 2ms after the modification. Please refer to https://github.com/pandas-dev/pandas/issues/38340 if you are interested in the performance tests.

Sure I could add other EAs here, will do that later. Thanks Andrew~

Reply:
I found ExtensionArray does not has the '_data' member, and only subclass of `BaseMaskedArray` has the _data member, so I guess the class below should be considered:
`FloatingArray`
`IntegerArray`
`BooleanArray`

``